pgmpy/readwrite/XMLBIF.py

```diff
@@ -303,10 +316,22 @@ def get_states(self):
             outcome_tag[var] = []
             for state in [State(var, state) for state in range(cpd.get_cardinality([var])[var])]:
                 state_tag = etree.SubElement(self.variables[var], "OUTCOME")
-                state_tag.text = str(state.state)
+                state_tag.text = self._make_valid_state(state.state)
                 outcome_tag[var].append(state_tag)
         return outcome_tag
 
+    def _make_valid_state(self, state):
+        """Transform the input statename into a valid state in XMLBIF. 
+        XMLBIF states must start with a letter an only contain letters,
+        numbers and underscores.
+        """
+        s = str(state)
+        s_allowed = re.sub("[^a-zA-Z0-9]", "_", s)
```

Comment:
@ankurankan What  are your views ? Should we use pyparsing or regular expression at here. This expression is quite simple to understand.